src/firewall.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The runner firewall manager."""
+import dataclasses
+import ipaddress
+import json
+import typing
+
+import yaml
+
+from utilities import execute_command
+
+
+@dataclasses.dataclass
+class FirewallEntry:
+    """Represent an entry in the firewall.
+
+    Attrs:
+        ip_range: The IP address range using CIDR notation.
+    """
+
+    ip_range: str
+
+    @classmethod
+    def decode(cls, entry: str) -> "FirewallEntry":
+        """Decode a firewall entry from a string.
+
+        Args:
+            entry: The firewall entry string, e.g. '192.168.0.1:80' or '192.168.0.0/24:80-90:udp'.
```

Comment:
Shouldn't the description be adapted ? Only CIDR notation is supported

Reply:
Fixed, thanks!